server/stream.go

```diff
@@ -3439,13 +3457,36 @@ func (mset *stream) subscribeToStream() error {
 	}
 	// Check if we need to setup mirroring.
 	if mset.cfg.Mirror != nil {
-		if err := mset.setupMirrorConsumer(); err != nil {
-			return err
+		// setup the initial mirror sourceInfo
+		mset.mirror = &sourceInfo{name: mset.cfg.Mirror.Name}
+		mirror := mset.mirror
```

Comment:
Just use mset.mirror below, will be clearer.

Reply:
Done